plugin/procfs/procfs.cc

```diff
@@ -0,0 +1,475 @@
+                        field_name.c_str()) == 0)
+      {
+        char    buff[1024];
+        String *res;
+        String  filter(buff, sizeof(buff), system_charset_info);
+        if (like_func->arguments()[1] != NULL &&
+            (res= like_func->arguments()[1]->val_str(&filter)))
+        {
+          like_arg->append(res->c_ptr_safe(), res->length());
+          return false;
+        }
+      }
+    }
+  }
+  return true;
+}
+
+
+static bool get_in_condition_argument(Item *                       cond,
+                                      std::map<std::string, bool> &in_args,
+                                      const std::string &          field_name)
+{
+  if (cond != 0 && cond->type() == Item::FUNC_ITEM)
+  {
+    Item_func *func= static_cast<Item_func *>(cond);
+    if (func != NULL && func->functype() == Item_func::IN_FUNC)
+    {
+      Item_func_in *in_func= static_cast<Item_func_in *>(func);
+      if (in_func->arguments()[0]->type() == Item::FIELD_ITEM &&
+          my_strcasecmp(system_charset_info,
+                        in_func->arguments()[0]->full_name(),
+                        field_name.c_str()) == 0)
+      {
+        char    buff[1024];
+        String *res;
+        String  filter(buff, sizeof(buff), system_charset_info);
+        for (uint i= 1; i < in_func->arg_count; ++i)
+        {
+          if (in_func->arguments()[i] != NULL &&
+              (res= in_func->arguments()[i]->val_str(&filter)) &&
+              res->length() > 0)
+          {
+            in_args[std::string(res->c_ptr_safe(), res->length())]= true;
+          }
+        }
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+static void fill_procfs_view_row(THD *thd, TABLE *table, const char *fname,
+                                 char *buf, size_t sz)
+{
+  if (sz == 0)
+    return;
+
+  table->field[0]->store(fname, strlen(fname), system_charset_info);
+  table->field[1]->store(buf, sz, system_charset_info);
+  schema_table_store_record(thd, table);
+}
+
+static std::vector<std::string> limited_glob_files(std::string path,
+                                                   std::string pattern,
+                                                   int         max_results)
+{
+  std::vector<std::string> files_found;
+
+  if (max_results <= 0)
+    return files_found;
+
+  DIR *          dir= opendir(path.c_str());
+  struct dirent *dir_entry;
+
+  if (!dir)
+    return files_found;
+
+  std::vector<char> real_file_path(PATH_MAX);
+
+  while ((dir_entry= readdir(dir)) != 0)
+  {
+    if (!fnmatch(pattern.c_str(), dir_entry->d_name,
+                 FNM_FILE_NAME | FNM_CASEFOLD | FNM_NOESCAPE | FNM_PERIOD))
+    {
+      if (dir_entry->d_type != DT_DIR)
+      {
+        std::string fname=
+            std::string(path).append("/").append(dir_entry->d_name);
+
+        if (realpath(fname.c_str(), real_file_path.data()))
+        {
+          if (strncmp("/sys/", real_file_path.data(), strlen("/sys/")) == 0 ||
```

Comment:
Either create a simple named constans for the lengths, or you could keep a `static const sys_len = strlen("/sys/")` outside the loop, to avoid repeated strlen calls. Same for the next line.